rust/rope/src/delta.rs

```diff
@@ -341,6 +336,60 @@ impl<'a, N: NodeInfo + 'a> Transformer<'a, N> {
     }
 }
 
+/// A builder for creating new `Delta` objects.
+///
+/// Note that all edit operations must be sorted; the start point of each
+/// interval must be no less than the end point of the previous one.
+pub struct Builder<N: NodeInfo> {
+    delta: Delta<N>,
+    last_offset: usize,
+}
+
+impl<N: NodeInfo> Builder<N> {
+    /// Creates a new builder, applicable to a base rope of length `base_len`.
+    pub fn new(base_len: usize) -> Builder<N> {
+        Builder {
+            delta: Delta {
+                els: Vec::new(),
+                base_len: base_len,
+            },
+            last_offset: 0,
+        }
+    }
+
+    /// Deletes the given interval. Panics if interval is not properly sorted.
+    pub fn delete(&mut self, interval: Interval) {
+        let (start, end) = interval.start_end();
```

Comment:
What would Range buy you that Interval doesn't? Interval does have the weirdness of supporting both open and closed endpoints, but I'm not sure that justifies having a whole nother type.